### PR TITLE
Feature/advanced search

### PR DESCRIPTION
**Pull request by Adamik10**
#### Link to issue

https://reload.atlassian.net/browse/DDFSOEG-533

#### Description

In this PR we add the advanced search app, it's data props, and update the dpl-design-system, and dpl-react libraries.

#### Screenshot of the result

-

#### Checklist

- [ ] My complies with [our coding guidelines](../docs/code_guidelines.md).
- [ ] My code passes our static analysis suite. If not then I have added a comment explaining why this change should be exempt from the code standards and process.
- [ ] My code passes our continuous integration process. If not then I have added a comment explaining why this change should be exempt from the code standards and process.

#### Additional comments or questions

The libraries still need to be properly updated when the PRs in [dpl design system](https://github.com/danskernesdigitalebibliotek/dpl-design-system/pull/284) and [dpl-react](https://github.com/danskernesdigitalebibliotek/dpl-react/pull/553) are approved and merged in.

### CODE REVIEW

File: composer.json [hunk 1]
```diff
@@ -122,15 +122,33 @@
...
+            ],
+            "drush/Commands/contrib/{$name}": [
+                "type:drupal-drush"
+            ],
+            "web/modules/custom/{$name}": [
+                "type:drupal-custom-module"
+            ],
+            "web/themes/custom/{$name}": [
+                "type:drupal-custom-theme"
+            ]
```
Comment (spaceo) [inline composer.json:151]: Apparently your linter/formatter (prettier?) wanted to format this, but other devs have been editing the file where this change was not applied. I am worried that we have a discrepancy here, although it's not a biggie...

File: cypress/e2e/withMappings/advanced-search.cy.ts [hunk 2]
```diff
@@ -0,0 +1,36 @@
...
+    cy.getBySel("search-header-dropdown").click();
+    cy.url().should("include", "advanced-search");
+    cy.get("h1").should("contain", "advanced search");
+  });
+
+  it("Can fill out the search form, translate it into CQL & switch to CQL search with the same translation", () => {
+    cy.visit("/advanced-search");
+    cy.getBySel("advanced-search-header-row").first().click().type("Harry");
+    cy.getBySel("advanced-search-header-row").eq(1).click().type("Prince");
+    cy.getBySel("advanced-search-header-row")
```
Comment (spaceo) [inline cypress/e2e/withMappings/advanced-search.cy.ts:13]: Great with all the cy selectors ❤️

File: web/modules/custom/dpl_react_apps/dpl_react_apps.module [hunk 3]
```diff
@@ -43,6 +44,8 @@ function dpl_react_apps_preprocess_page(array &$variables): void {
         'string-suggestion-author-text' => t('Author', [], ['context' => 'Search Header']),
         'string-suggestion-topic-text' => t('Topic', [], ['context' => 'Search Header']),
         'string-suggestion-work-text' => t('Work', [], ['context' => 'Search Header']),
+        'header-dropdown-item-advanced-search-text' => t('Advanced search', [], ['context' => 'Search Header']),
+        'search-header-dropdown-text' => t('Dropdown with additional search functions', [], ['context' => 'Search Header']),
       // Add external API base urls.
```
Comment (spaceo) [inline web/modules/custom/dpl_react_apps/dpl_react_apps.module:49]: I reccomend ordering the text props alphabetically. It helps the further maintenance of it.

File: web/modules/custom/dpl_react_apps/src/Controller/DplReactAppsController.php [hunk 4]
```diff
@@ -158,6 +159,100 @@ public function search(): array {
...
+      'advanced-search-filter-ebook-text' => $this->t('ebook', [], ['context' => 'advanced search']),
+      'advanced-search-filter-audio-book-text' => $this->t('audio book', [], ['context' => 'advanced search']),
+      'advanced-search-filter-article-text' => $this->t('article', [], ['context' => 'advanced search']),
+      'advanced-search-filter-movie-text' => $this->t('movie', [], ['context' => 'advanced search']),
+      'advanced-search-filter-music-text' => $this->t('music', [], ['context' => 'advanced search']),
+      'advanced-search-filter-physical-text' => $this->t('physical', [], ['context' => 'advanced search']),
+      'advanced-search-filter-online-text' => $this->t('online', [], ['context' => 'advanced search']),
+      'advanced-search-filter-fiction-text' => $this->t('fiction', [], ['context' => 'advanced search']),
+      'advanced-search-filter-non-fiction-text' => $this->t('Non-fiction', [], ['context' => 'advanced search']),
+      'advanced-search-filter-all-text' => $this->t('All', [], ['context' => 'advanced search']),
```
Comment (spaceo) [inline web/modules/custom/dpl_react_apps/src/Controller/DplReactAppsController.php:241]: Wow. A lot of strings 😅 Great work.
 I recommend sorting the text props alphabetically to help the maintenance of them in the future.

**Review by spaceo — APPROVED** [review]
Looks good 👍 
I have just a few minor comments.
I guess this has to wait with being merged and having the dpl-react dep pointed to dev until the dpl-react changes have been merged?

File: composer.json (hunk 1 above)
Comment (Adamik10) [reply composer.json:151]: If we just have to do it once, maybe it's okay? Let's keep an eye on this in the future 🙈

**Pull request merged**